website_sale_qty/static/src/js/website_sale_qty.js

```diff
@@ -0,0 +1,41 @@
+/* Copyright 2016-2017 LasLabs Inc.
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html) */
+
+odoo.define('website_sale_qty', function(require) {
+    'use strict';
+
+    var base = require('web_editor.base');
+    base.ready().done(function() {
+        var radios = $('.js-pqt-input');
+
+        if (radios.length > 0) {
+            var radioMap = {};
+            _.each(radios, function(element) {
+                var $element = $(element);
+                radioMap[$element.val()] = $element;
+            });
+            var quantityInput = $('.css_quantity input');
+            var quantityToggles = $('.css_quantity a');
+
+            radios.click(function(event) {
+                var newValue = $(event.currentTarget).val();
+                quantityInput.val(newValue).trigger('change');
+            });
+
+            var updateRadios = function() {
+                radios.prop('checked', false);
+
+                var newValue = quantityInput.val();
+                if (newValue in radioMap) {
+                    radioMap[newValue].prop('checked', true);
+                }
+            };
+
+            quantityInput.on('input', updateRadios);
+            quantityToggles.click(function() {
+                // This allows website_sale.website_sale to update the input first
```

Comment:
> My understanding is that there are no race conditions in JS

A race condition can exist in anything. It is simply when one process depends upon something from another without an explicit sequential chain between the two. They're actually more common in JS due to the non-sequential nature of it.

Using a `setTimeout` is typically a trick that developers will use to solve race conditions. To put it into a simple example, let's pretend that the triggers for page load completion don't exist (or we don't know about them). We have a method that depends on something to be in the DOM, but the DOM isn't ready for a bit. A nasty hack would be to add a `setTimeout` of some interval that works on the developer's computer, but may not work on other slower or heavily loaded machines. It's a newb hack really, and is something I explicitly look for when reviewing JS.

In your instance you're actually doing an expert hack. The `setTimeout` of `0` is only guaranteeing that your operation is shifted to the bottom of the event queue, of which the `website_sale.website_sale` event actions are also a part of. Because those actions already exist in the queue, you are guaranteeing a solution no matter the environment.

This is alluded to in your comment, I was just suggesting that you possibly mention why it works so that someone doesn't come and try to re-engineer for some reason.